src/Stats.AzureCdnLogs.Common/AzureHelpers/AzureBlobLeaseManager.cs

```diff
@@ -19,96 +23,105 @@ public class AzureBlobLeaseManager
+        /// In order to stop the renew task the <see cref="Stats.AzureCdnLogs.Common.AzureBlobLeaseManager.TryReleaseLockAsync(AzureBlobLockResult)"/> needs to be invoked
         /// or the token to be cancelled.
         /// </summary>
         /// <param name="blob">The blob to acquire the lease on.</param>
-        /// <param name="token">A token to cancel the operation.</param>
-        /// <param name="renewStatusTask">The renew task.</param>
-        /// <returns>True if the lease was acquired. </returns>
-        public AzureBlobLockResult AcquireLease(CloudBlob blob, CancellationToken token)
+        /// <returns>An <see cref="AzureBlobLockResult"/> indicating the result of the lease acquisition. 
+        /// If the lease is successfully acquired, the result will contain the lease ID and a cancellation token 
+        /// source that can be used to stop the lease renewal task.</returns>
+        public async Task<AzureBlobLockResult> AcquireLease(BlobClient blob)
         {
-            blob.FetchAttributes();
-            if (token.IsCancellationRequested || blob.Properties.LeaseStatus == LeaseStatus.Locked)
+            try
             {
-                _logger.LogInformation("AcquireLease: The operation was cancelled or the blob lease is already taken. Blob {BlobUri}, Cancellation status {IsCancellationRequested}, BlobLeaseStatus {BlobLeaseStatus}.",
-                    blob.Uri.AbsoluteUri,
-                    token.IsCancellationRequested,
-                    blob.Properties.LeaseStatus);
-                return AzureBlobLockResult.FailedLockResult(blob);
-            }
-            var proposedLeaseId = Guid.NewGuid().ToString();
-            var leaseId = blob.AcquireLease(TimeSpan.FromSeconds(MaxRenewPeriodInSeconds), proposedLeaseId);
-            var lockResult = new AzureBlobLockResult(blob: blob, lockIsTaken: true, leaseId: leaseId, linkToken: token);
+                var leaseClient = blob.GetBlobLeaseClient();
+                var leaseResponse = await leaseClient.AcquireAsync(TimeSpan.FromSeconds(MaxRenewPeriodInSeconds));
+                /*if (leaseResult.IsSuccess)
+                {
+                    _logger.LogInformation("AcquireLease: Lease was acquired for BlobUri {BlobUri}.", blob.Uri.AbsoluteUri);
+                }
+                else
+                {
+                    _logger.LogInformation("AcquireLease: The operation was cancelled or the blob lease is already taken. Blob {BlobUri}.", blob.Uri.AbsoluteUri);
+                    return AzureBlobLockResult.FailedLockResult(blob);
+                }*/
 
-            //start a task that will renew the lease until the token is cancelled or the Release methods was invoked
-            var renewStatusTask = new Task( (lockresult) =>
+                string leaseId = leaseResponse.Value.LeaseId;
+                var lockResult = new AzureBlobLockResult(blob, true, leaseId, CancellationToken.None);
```

Comment:
It's nice to pass parameter name if it's not obvious.
```suggestion
                var lockResult = new AzureBlobLockResult(blob, lockIsTaken : true, leaseId, CancellationToken.None);
```